e2e/forwarder/service.go

```diff
@@ -81,6 +81,17 @@ func (s *Service) WithProtocol(protocol string) *Service {
 	return s
 }
 
+func (s *Service) SelfSigned() *Service {
+	delete(s.Environment, "FORWARDER_CERT_FILE")
+	delete(s.Environment, "FORWARDER_KEY_FILE")
```

Comment:
I'm not a fan of this approach. It forces user to keep the right order of options, otherwise it can break. However, when used with caution, maybe it's not that bad.

Reply:
```
				AddService(
					forwarder.UpstreamProxyService().
						WithSelfSigned("https")).
```

Reply:
It's definitely better. It would be perfect if there was an assertion, but it is not necessary.